src/fsharp/fsc.fs

```diff
@@ -1491,22 +1491,15 @@ module StaticLinker =
                         | Some provAssemStaticLinkInfo -> yield (importedBinary,provAssemStaticLinkInfo) ]
 #endif
         if tcConfig.compilingFslib && tcConfig.compilingFslib20.IsSome then 
-            (fun ilxMainModule -> LegacyFindAndAddMscorlibTypesForStaticLinkingIntoFSharpCoreLibraryForNet20 (tcConfig, ilGlobals, ilxMainModule))
+            [], (fun ilxMainModule -> LegacyFindAndAddMscorlibTypesForStaticLinkingIntoFSharpCoreLibraryForNet20 (tcConfig, ilGlobals, ilxMainModule))
```

Comment:
The static linker is calculated closer to the beginning of the compile chain so that the provided type definitions are available for the type check phases, the static linker is passed through the compile chain and then executed in `main3`.  Theres no real change in code just the compile phase the static linker is calculated.